common/types.go

```diff
@@ -22,20 +21,32 @@ func NewBigInt(v int64) BigInt {
 }
 
 func (b BigInt) MarshalText() ([]byte, error) {
-	return []byte(b.String()), nil
+	return b.Int.MarshalText()
 }
 
 func (b *BigInt) UnmarshalText(text []byte) error {
 	return b.Int.UnmarshalText(text)
 }
 
 func (b BigInt) MarshalJSON() ([]byte, error) {
-	return []byte(fmt.Sprintf(`"%s"`, b.String())), nil
+	t, err := b.MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(string(t))
 }
 
 func (b *BigInt) UnmarshalJSON(text []byte) error {
-	v := strings.Trim(string(text), "\"")
-	return b.Int.UnmarshalJSON([]byte(v))
+	var s string
+	err := json.Unmarshal(text, &s)
+	if err != nil {
+		return err
+	}
+	return b.UnmarshalText([]byte(s))
+}
+
+func (b BigInt) String() string {
+	return b.Int.String()
```

Comment:
Nit/curiosity: In https://github.com/oasisprotocol/nexus/issues/566, I suggested `(&b.Int).String()`, because we need a `*big.Int` to call `String()` on it, whereas `b.Int` is of type `big.Int`. In my understanding, the code as written shouldn't work, yet it does. Do you know why?

Reply:
doesn't it usually automatically use pointer receivers? let me see if I can find a reference for this

Reply:
https://go.dev/ref/spec#Calls

> If x is [addressable](https://go.dev/ref/spec#Address_operators) and &x's method set contains m, x.m() is shorthand for (&x).m()

thanks go :skull: